llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7035,6 +7035,63 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
       return new ICmpInst(I.getSwappedPredicate(Pred), Builder.CreateXor(X, Y),
                           Z);
 
+    // Transform X s<  ~X  -->   X s<  0
+    //           X s>  ~X  -->   X s>  -1
+    //           X s>= ~X  -->   X s>  ~X
+    //           X s<= ~X  -->   X s<  ~X
```

Comment:
```suggestion
    //           X s>= ~X  -->   X s>  -1
    //           X s<= ~X  -->   X s<  0
```
Specify final result.

Reply:
I have a question. 

if result of code I added are different with result of PASS, which result is better to comment?

example. in my code, `X u<= ~X` transform to `X u>  -128` but PASS final result is `X s> -1`

Reply:
I think we should emit `X s> -1` directly.